pricelist_items_auto_update_date/models/pricelist_items.py

```diff
@@ -32,8 +32,9 @@ def create(self, vals_list):
         vals_list_old = vals_list.copy()
         for values in vals_list_old:
             if values.get('date_start'):
-                for item in self.pricelist_id.item_ids.search([('pricelist_id', '=', values['pricelist_id']), ('product_tmpl_id', '=', values['product_tmpl_id'])], order='date_start asc'):
-                    date_min = self.pricelist_id.item_ids.search([('pricelist_id', '=', values['pricelist_id']), ('product_tmpl_id', '=', values['product_tmpl_id'])], order='date_start asc')[0].date_start
+                pricelist_item_ids = self.search([('pricelist_id', '=', values['pricelist_id']), ('product_tmpl_id', '=', values['product_tmpl_id'])], order='date_start asc')
+                date_min = pricelist_item_ids[0].date_start
```

Comment:
This piece of code is now outside for statement, at least one element is not guaranteed:
```suggestion
                date_min = pricelist_item_ids and pricelist_item_ids[0].date_start
```